pallets/wnodl/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/v3/runtime/frame>
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+    pub use ethereum_types::Address as EthAddress;
+    use frame_support::{
+        pallet_prelude::*,
+        traits::{Contains, Currency, LockableCurrency},
+    };
+    use frame_system::pallet_prelude::*;
+    pub use sp_core::H256 as EthTxHash;
+    use sp_runtime::traits::{CheckedAdd, Saturating, Zero};
+
+    pub type CurrencyOf<T> = <T as Config>::Currency;
+    pub type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Trusted bots/oracles which can settle funds after wrapping is initiated
+        type Oracles: Contains<Self::AccountId>;
+
+        /// The customers who've gone under the KYC process and are eligible to wrap their Nodls.  
+        type KnownCustomers: Contains<Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_initiated)]
+    /// The sum of wNodl funds that is initiated by this pallet so far.
+    pub type TotalInitiated<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_settled)]
+    /// The sum of wNodl funds that is settled by this pallet so far.
+    pub type TotalSettled<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn balances)]
+    /// The amount of initiated and settled wnodl for an account id.
```

Comment:
Minor: should we escape `wNODL` here?

```suggestion
/// The amount of initiated and settled `wNODL` for an account id.
```

Reply:
Done